assets/js/state/sagas/activityLogsSettings.js

```diff
@@ -0,0 +1,45 @@
+import { get } from 'lodash';
+import { put, call, takeEvery, debounce } from 'redux-saga/effects';
+import { getSettings, updateSettings } from '@lib/api/activityLogsSettings';
+
+import {
+  FETCH_ACTIVITY_LOGS_SETTINGS,
+  UPDATE_ACTIVITY_LOGS_SETTINGS,
+  startLoadingActivityLogsSettings,
+  setActivityLogsSettings,
+  setActivityLogsSettingsErrors,
+  setEditingActivityLogsSettings,
+  setNetworkError,
+} from '@state/activityLogsSettings';
+
+export function* fetchActivityLogsSettings() {
+  yield put(startLoadingActivityLogsSettings());
+
+  try {
+    const response = yield call(getSettings);
+    yield put(setActivityLogsSettings(response.data));
+  } catch (error) {
+    const errorCode = get(error, ['response', 'status']);
+    if (errorCode !== 403) {
+      yield put(setNetworkError(true));
+    }
+  }
+}
+
+export function* updateActivityLogsSettings({ payload }) {
+  yield put(startLoadingActivityLogsSettings());
+  try {
+    const response = yield call(updateSettings, payload);
+    yield put(setActivityLogsSettings(response.data));
+    yield put(setEditingActivityLogsSettings(false));
+    yield put(setActivityLogsSettingsErrors([]));
+  } catch (error) {
+    const errors = get(error, ['response', 'data', 'errors'], []);
```

Comment:
We might be missing the opportunity to give the user a proper feedback on generic errors that might not fall into validation ones or that do not satisfy the expectation of a specific shape in the response body.

I forced the update endpoint to return 500, and I think it might be valuable to provide some feedback like `unable to save retention time` (or whatever message).

https://github.com/trento-project/web/assets/8167114/ebc9ac35-a06a-43b1-94c0-54c05d99d9ae

I am afraid a non-feedback might be misleading.

Reply:
Good point. We can add a label aside the saving buttons, for global errors.

Reply:
Added generic error in https://github.com/trento-project/web/pull/2727/commits/09bea2e8f22d2bf2f3432b3e6bd4af0f0664970d.

The UI simply shows the message `Something went wrong while saving` when an unstructured error appears (see storybook). We might provide a more descriptive message if we can add some intelligence to the error parsing, though..

Reply:
Improved in https://github.com/trento-project/web/pull/2727/commits/52303cac574bda7f2b54edbb8574cd63f3089ca2